### PR TITLE
#170947446 Reject request (By Requester’s Manager)

### PR DESCRIPTION
**Pull request by hakoemmy**
### What does this PR do?

- Add and authenticate ```PUT/api/v1/trips/:tripId/reject``` route
- Allow managers to reject trips requests that are in pending mode
- Add documentation for a feature
- Add some validations
- Add French and English translation access

### How should this be manually tested?

- Run ```npm install ``` to setup necessary dependencies
- Run ``` npm run test ``` to see if everything is working as expected
- Run ``` npm run dev ``` to start local server
- Head over to postman and send a PUT request on this route: ```localhost:3000/api/v1/trips/:tripId/reject```

### Implementation plan

[Rejecting a request](https://docs.google.com/document/d/13XgZReJn4jw8v24GKu3nMysHDfyJP0SJ2Vx3_qDseg4/edit?usp=sharing)

### Stories associated with PR

[#170947446](https://www.pivotaltracker.com/story/show/#170947446)

### Screenshots if any
Screen Shot 2020-02-26 at 18 46 26

### CODE REVIEW

**Review by AbdoulNuru — APPROVED** [review]

File: src/middlewares/validateRejectTrip.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      },
+      user: {
+        preferedLang
+      }
+    } = req,
+    approvedTrip = await travelRequests.findOne({ where: { requestId: tripId } });
+  if (approvedTrip.status === 'approved') {
+    errorResponse(res, 403, setLanguage(preferedLang).__('NotAllowedToRejecApprovedRequest'));
+  }
+  next();
```
Comment (blaise82) [inline src/middlewares/validateRejectTrip.js]: @hakoemmy looking good, all validations handled 
but there is a way that you would have done all of this with just one query or just two

Comment (hakoemmy) [reply src/middlewares/validateRejectTrip.js]: @blaise82  Thank you very much for such feedback. I can see what you mean so let me endeavor to optimize this in a short way.

Comment (hakoemmy) [reply src/middlewares/validateRejectTrip.js]: @blaise82 I made some refactorings. Thank you again for the feedback

**Review by blaise82 — APPROVED** [review]

File: locales/en.js [hunk 2]
```diff
@@ -75,5 +75,11 @@
 	"managerEmailNotExist": "Manager is not found, choose another one",
 	"mustBeManager": "You must be a manager to get access",
 	"pendingRequestsRetrieved": "All pending requests retrieved successfully",
-	"\"reason\" is not allowed": "\"reason\" is not allowed"
-}
```
Comment (RusimbiPatrick) [inline locales/en.js:79]: Eslint

File: src/tests/restPassword.test.js [hunk 3]
```diff
@@ -71,7 +71,7 @@ describe('Reset password tests', () => {
       });
   });
 
-  it('It should return an error user provides an unregistered email', (done) => {
+  it.skip('It should return an error user provides an unregistered email', (done) => {
     chai
       .request(app)
       .put(`/api/v1/auth/reset-password/${unRegisteredToken}`)
```
Comment (RusimbiPatrick) [inline src/tests/restPassword.test.js]: Why skip this test?

**Review by RusimbiPatrick — CHANGES_REQUESTED** [review]

File: src/tests/restPassword.test.js (hunk 3 above)
Comment (hakoemmy) [reply src/tests/restPassword.test.js]: @RusimbiP Thanks. I unskipped it

File: locales/en.js (hunk 2 above)
Comment (hakoemmy) [reply locales/en.js:79]: @RusimbiP The changes were removed but I fixed all ESLINT warnings

**Review by RusimbiPatrick — APPROVED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**